runtests/mpi/tester.py

```diff
@@ -74,15 +83,19 @@ def create_comm(size):
 
     return communicators[size], color
 
-def MPITestFixture(commsize, scope='function'):
+def MPITestFixture(commsize, scope='function', mpi_missing='fail'):
     """ Create a test fixture for MPI Communicators of various commsizes """
 
     @pytest.fixture(params=commsize, scope=scope)
     def fixture(request):
-        from mpi4py import MPI
-        MPI.COMM_WORLD.barrier()
         try:
-            comm, color = create_comm(request.param)
+            from mpi4py import MPI
+            MPI.COMM_WORLD.barrier()
```

Comment:
Limit the scope of the try block to import only.

Reply:
Sorry I am not sure I understand. 

Do you mean 

```
try:
            from mpi4py import MPI
            
        except ImportError:
            if mpi_missing != 'ignore':
                raise
MPI.COMM_WORLD.barrier()
```

for mpi_missing = ignore the barrier will fail.

Reply:
More or less, I wonder if you can do 
```
_MPI4PY_UNSUPPORTED = False
try:
  from mpi4py import MPI
except ImportError:
  _MPI4PY_UNSUPPORTED = True
```
Then use _MPI4PY_UNSUPPORTED to branch in the rest of mpi/tester.py.
```
if not _MPI4PY_UNSUPPORTED:
  MPI.COMM_WORLD.barrier()
```
It's less smart use of try catch, but an upper case symbol like (`_MPI4PY_UNSUPPORTED`) will help us to track the specializations.

Reply:
Actually we check `MPI is not None` in the rest of the code.

What about 

```python
    try:
        from mpi4py import MPI
    except ImportError:
        if mpi_missing != 'ignore':
            raise
        MPI = None

if MPI is not None: 
    MPI.COMM_WORLD.barrier()
```

Reply:
Sure!

Reply:
addressed in 624497a9